File: Project_Three/spring-project/entity-management/src/main/java/com/ozalp/entity_management_spring/business/impls/PersonServiceImpl.java
package com.ozalp.entity_management_spring.business.impls;

import com.ozalp.entity_management_spring.business.mappers.PersonMapper;
import com.ozalp.entity_management_spring.business.service.PersonService;
import com.ozalp.entity_management_spring.dataAccess.PersonRepository;
import com.ozalp.entity_management_spring.models.dtos.requests.CreatePersonRequest;
import com.ozalp.entity_management_spring.models.dtos.responses.PersonResponse;
import com.ozalp.entity_management_spring.models.entities.Person;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
@RequiredArgsConstructor
public class PersonServiceImpl implements PersonService {

    private final PersonRepository repository;
    private final PersonMapper mapper;

    @Override
    public void add(CreatePersonRequest request) {
        Person reqEntity = mapper.toEntity(request);
        repository.save(reqEntity);
    }

    @Override
    public void delete(String name) {
        List<Person> myEntityList = repository.findByName(name);

        for (Person entity : myEntityList) {
            repository.delete(entity);
        }
    }

    @Override
    public List<PersonResponse> getAll() {
        return repository.findAll()
                .stream()
                .map(mapper::toResponse)
                .toList();
    }

    @Override
    public List<PersonResponse> search(String name) {
        return repository.findByName(name).stream()
                .map(mapper::toResponse)
                .toList();
    }
}

File: Project_Two/EntityManagementAspNet/EntityManagementAspNet.Business/Impl/PersonServiceImpl.cs
using EntityManagementAspNet.Business.Dtos;
using EntityManagementAspNet.Business.Service;
using EntityManagementAspNet.DataAccess.Abstract;
using EntityManagementAspNet.Entity;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace EntityManagementAspNet.Business.Impl
{
    public class PersonServiceImpl : IPersonService
    {

        private IPersonDal _personDal;

        public PersonServiceImpl(IPersonDal personDal)
        {
            _personDal = personDal;
        }

        public void add(CreatePersonRequest request)
        {
            var person = new Person
            {
                Name = request.Name,
                Surname = request.Surname,
                Email = request.Email,
                PhoneNumber = request.PhoneNumber,
                Age = request.Age
            };
            _personDal.Add(person);
        }

        public void delete(string name)
        {
            var person = _personDal.GetAll(p => p.Name == name).FirstOrDefault();
            if (person != null)
            {
                _personDal.Delete(person);
            }
        }

        public List<PersonResponse> getAll()
        {
            var persons = _personDal.GetAll();
            return persons.Select(p => new PersonResponse
            {
                Id = p.Id,
                Name = p.Name,
                Surname = p.Surname,
                Email = p.Email,
                PhoneNumber = p.PhoneNumber,
                Age = p.Age

            }).ToList();
        }

        public List<PersonResponse> search(string name)
        {
            var persons = _personDal.GetAll(p => p.Name.Contains(name));
            return persons.Select(p => new PersonResponse
            {
                Id = p.Id,
                Name = p.Name,
                Surname = p.Surname,
                Email = p.Email,
                PhoneNumber = p.PhoneNumber,
                Age = p.Age
            }).ToList();
        }
    }
}
